community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/util.py

```diff
@@ -1928,26 +1932,58 @@ def nodeset_map(self, hostnames: list):
             nodeset_map[self.node_nodeset_name(node)].append(node)
         return nodeset_map
 
+    def _get_job_info(self, job_info, job_id=None) -> Optional[Job]:
+        """Extract job details"""
+        if job_id is None:
```

Comment:
It's not clear what the point of `job_id` here.
If `job_info` contains different `JobId` it would be just silently overwritten.  
My advice would be to either get rid of `job_id` argument, or add validation that `JobId` matches `job_id`